ddapm_test_agent/trace.py

```diff
@@ -702,6 +702,354 @@ class TraceChunkV07(TypedDict):
+        return value
+    elif isinstance(value, int):
+        if value >= len(string_table) or value < 0:
+            raise ValueError(f"Value {value} is out of range for string table of length {len(string_table)}")
+        return string_table[value]
+
+
+def _convert_v1_payload(data: Any) -> v04TracePayload:
+    if not isinstance(data, dict):
+        raise TypeError("Trace payload must be a map, got type %r." % type(data))
+
+    string_table: List[str] = [""]  # 0 is reserved for empty string
+
+    v04Payload: List[List[Span]] = []
+
+    for k, v in data.items():
+        if k == 1:
+            raise TypeError("Message pack representation of v1 trace payload must stream strings")
+        elif k > 1 and k < 10:  # All keys from 2-9 are strings, for now we can just build the string table
+            # TODO: In the future we can assert on these keys
+            if isinstance(v, str):
+                string_table.append(v)
+        elif k == 11:
+            if not isinstance(v, list):
+                raise TypeError("Trace payload 'chunks' (11) must be a list.")
+            for chunk in v:
+                v04Payload.append(_convert_v1_chunk(chunk, string_table))
+        else:
+            raise TypeError("Unknown key %r in v1 trace payload" % k)
+    return cast(v04TracePayload, v04Payload)
+
+
+def _convert_v1_chunk(chunk: Any, string_table: List[str]) -> List[Span]:
+    if not isinstance(chunk, dict):
+        raise TypeError("Chunk must be a map.")
+
+    priority, origin, decision_maker = "", "", ""
+    trace_id, trace_id_high = 0, 0
+    meta: Dict[str, str] = {}
+    metrics: Dict[str, MetricType] = {}
+    for k, v in chunk.items():
+        if k == 1:
+            priority = v
+        elif k == 2:
+            origin = _get_and_add_string(string_table, v)
+        elif k == 3:
+            if not isinstance(v, list):
+                raise TypeError("Chunk Attributes must be a list, got type %r." % type(v))
+            _convert_v1_attributes(v, meta, metrics, string_table)
+        elif k == 4:
+            if not isinstance(v, list):
+                raise TypeError("Chunk 'spans'(4) must be a list.")
+            spans: List[Span] = []
+            for span in v:
+                converted_span = _convert_v1_span(span, string_table)
+                spans.append(converted_span)
+        elif k == 5:
+            raise TypeError("Tracers must not set the droppedTrace(5) flag.")
+        elif k == 6:
+            if len(v) != 16:
+                raise TypeError("Trace ID must be 16 bytes, got %r." % len(v))
+            # trace_id is a 128 bit integer in a bytes array, so we need to get the last 64 bits
+            trace_id = int.from_bytes(v[8:], "big")
+            trace_id_high = int.from_bytes(v[:8], "big")
+        elif k == 7:
+            decision_maker = _get_and_add_string(string_table, v)
+        else:
+            raise TypeError("Unknown key %r in v1 trace chunk" % k)
+
+    if len(spans) == 0:
```

Comment:
oh fun, I _think_ there are scenarios when we send a single empty trace as a test.... but I cannot remember where that is

Reply:
I don't think this is an issue at all, just calling it out, I _think_ we do/used to do something like this.